### PR TITLE
feat-461: remove duplicate functions

### PR DESCRIPTION
**Pull request by khendrikse**
I wanted to try to implement an idea by @eduardoboucas as [described here](https://github.com/netlify/edge-bundler/pull/359), this is that implementation.

### CODE REVIEW

**Review by eduardoboucas — COMMENTED** [review]
Sorry for the delay in reviewing this one!

Looking at the diff, I'm not sure that this simplifies things as much as I thought. There are a couple of things we could do, but I think we're quickly getting into bikeshedding territory here. 😅 

I think this approach is fine (except for that `ts-expect-error` comment I left), just as your existing approach is fine. So I'm happy to defer to you to pick the one you feel better about.

File: node/finder.ts [hunk 1]
```diff
@@ -8,25 +8,21 @@ import { nonNullable } from './utils/non_nullable.js'
...
-        return { ...acc, [name]: path }
-      }
-
-      return acc
-    }, {}),
-  ) as string[]
+export const hasPrecedingDuplicate = (path: string, functionExtensions: Map<string, number>) => {
+  const { ext, name } = parse(path)
+  const extIndex = ALLOWED_EXTENSIONS.indexOf(ext)
+  // @ts-expect-error functionExtensions might be empty
```
[the remark this answers is not in the document]
Comment (khendrikse) [reply node/finder.ts:14]: Thats what I told typescript 🤷‍♀️ but then it kept shouting at me :(

[the remark this answers is not in the document]
Comment (danez) [reply node/finder.ts:14]: Typescript has no knwoledge that `has() and get()` are related. on alternative is to do

```ts
const extensions = functionExtensions.get(name)
if (!extensions || extensions > extIndex) {
```

**Comment by khendrikse** [comment]
Seeing as the difference isn't too big, I'll be closing this for now :)

**Pull request closed without merging**